script.js

```diff
@@ -40,11 +48,15 @@ class ToDoList {
         this.saveToLocalStorage();
     }
     saveToLocalStorage() {
-        window.localStorage.setItem("to-do-list", JSON.stringify(this.toDos))
+        // Check if useLocalStorage is enabled for this instance, so we only save to
+        // local storage if the list requests it.
+        if (this.useLocalStorage) {
```

Comment:
`this.useLocalStorage` returns undefined because even though you pass it as a parameter to the constructor, it is not defined in the object. You need to assign it manually, so inside the constructor you'd need to do: `this.useLocalStorage = useLocalStorage`

Reply:
Gigantic error on my part, thank you for catching! This will teach me for not testing first 😛. Fixing.